jetty-websocket/websocket-jetty-server/src/main/java/org/eclipse/jetty/websocket/server/JettyWebSocketServerContainer.java

```diff
@@ -117,6 +120,13 @@ public static JettyWebSocketServerContainer ensureContainer(ServletContext servl
         addBean(sessionTracker);
     }
 
+    @Override
+    public void lifeCycleStopping(LifeCycle event)
+    {
+        contextHandler.removeBean(this);
+        contextHandler.removeEventListener(this);
```

Comment:
I was expecting that the stopping event would reference the same objects as the starting - eg what about the container that was added as a managed bean?

Reply:
`this` is the same container that was added as a managed bean.

Reply:
I think its probably clearer to do this removal from a separate `LifeCycle.Listener` added in `ensureContainer`, the same way it is done in `WebSocketServerComponents`. So I will make that change.